plugins/buildClusters/update.js

```diff
@@ -28,68 +28,39 @@ module.exports = () => ({
+            const adminDetails = request.server.plugins.banners.screwdriverAdminDetails(username, scmDisplayName);
 
-                if (!adminDetails.isAdmin) {
-                    return boom.forbidden(
-                        `User ${adminDetails.userDisplayName}
+            if (!adminDetails.isAdmin) {
+                return boom.forbidden(
+                    `User ${adminDetails.userDisplayName}
                         does not have Screwdriver administrative privileges.`
-                    );
-                }
-
-                return buildClusterFactory
-                    .list({
-                        params: {
-                            name,
-                            scmContext: userContext
-                        }
-                    })
-                    .then(buildClusters => {
-                        if (!Array.isArray(buildClusters)) {
-                            throw boom.badData('Build cluster list returned non-array.');
-                        }
-                        if (buildClusters.length === 0) {
-                            throw boom.notFound(
-                                `Build cluster ${name}, scmContext ${payload.scmContext} does not exist`
-                            );
-                        }
-
-                        Object.assign(buildClusters[0], request.payload);
-
-                        return buildClusters[0]
-                            .update()
-                            .then(updatedBuildCluster => h.response(updatedBuildCluster.toJson()).code(200));
-                    })
-                    .catch(err => {
-                        throw err;
-                    });
+                );
             }
-            // Must provide scmOrganizations if not a Screwdriver cluster
-            if (scmOrganizations && scmOrganizations.length === 0) {
+
+            // Must provide scmOrganizations if not a Screwdriver managed cluster
+            if (!managedByScrewdriver && scmOrganizations && scmOrganizations.length === 0) {
                 return boom.badData(`No scmOrganizations provided for build cluster ${name}.`);
             }
 
             return buildClusterFactory
                 .list({
                     params: {
                         name,
-                        scmContext: payload.scmContext
+                        scmContext: userContext
                     }
                 })
                 .then(buildClusters => {
                     if (!Array.isArray(buildClusters)) {
                         throw boom.badData('Build cluster list returned non-array.');
                     }
                     if (buildClusters.length === 0) {
-                        throw boom.notFound(`Build cluster ${name} scmContext ${payload.scmContext} does not exist`);
+                        throw boom.notFound(`Build cluster ${name}, scmContext ${payload.scmContext} does not exist`);
```

Comment:
```suggestion
                        throw boom.notFound(`Build cluster ${name}, scmContext ${userContext} does not exist`);
```

Reply:
but we are setting this: https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/buildClusters/create.js#L25